src/journal/views.py

```diff
@@ -1083,19 +1085,33 @@ def publish_article(request, article_id):
                 )
 
         if 'pubdate' in request.POST:
-            date_set, pubdate_errors = logic.handle_set_pubdate(
-                request,
-                article,
+            pub_date_form = submission_forms.PubDateForm(
+                request.POST,
+                instance=article,
             )
-            if not pubdate_errors:
-                return redirect(
-                    reverse(
-                        'publish_article',
-                        kwargs={'article_id': article.pk},
+            if pub_date_form.is_valid():
+                article = pub_date_form.save()
+                if article.date_published:
+                    messages.add_message(
+                        request, messages.SUCCESS,
+                        _(
+                            f'Publication date set to { article.date_published.strftime("%Y-%m-%d %H:%M %Z") } '
+                            f'({ naturaltime(article.date_published) })'
+                        )
+                    )
+                else:
+                    messages.add_message(
+                        request, messages.SUCCESS,
+                        _('Publication date unset')
                     )
-                )
             else:
-                modal = 'pubdate'
```

Comment:
Is this unset on purpose? When the form is invalid generally we'd want to open the modal again.